api/v1/sriovnetworknodeconfig_types.go

```diff
@@ -0,0 +1,58 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SriovNetworkNodeConfigSpec defines the desired state of SriovNetworkNodeConfig
+type SriovNetworkNodeConfigSpec struct {
+	// OvsHardwareOffloadConfig describes the OVS HWOL configuration for selected Nodes
+	OvsHardwareOffloadConfig OvsHardwareOffloadConfig `json:"ovsHardwareOffloadConfig,omitempty"`
+}
+
+type OvsHardwareOffloadConfig struct {
+	// Name is mandatory and must be unique.
+	// On Kubernetes:
+	// Name is the name of OvsHardwareOffloadConfig
+	// On OpenShift:
+	// Name is the name of MachineConfigPool to be enabled with OVS hardware offload
+	Name string `json:"name"`
+	// On Kubernetes:
+	// NodeSelector selects Kubernetes Nodes to be configured with OVS HWOL configurations
+	// OVS HWOL configurations are generated automatically by Operator
+	// Labels in NodeSelector are ANDed when selecting Kubernetes Nodes
+	// On OpenShift:
+	// NodeSelector matches on Labels defined in MachineConfigPoolSpec.NodeSelector
+	// OVS HWOL MachineConfigs are generated and applied to Nodes in MachineConfigPool
+	// Labels in NodeSelector are ANDed when matching on MachineConfigPoolSpec.NodeSelector
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
Shall we also add the logic to check the `NodeSelector` in the k8s_plugin for the Kubernetes scenario?